R/htmlwidgets.R

```diff
@@ -167,6 +167,12 @@ addHook <- function(x, hookName, jsCode, data = NULL) {
   x
 }
 
+do_use_aria <- function(knitrOptions) {
+  getOption("htmlwidgets.USE_ARIA") %||%
+  ("fig.alt" %in% names(knitrOptions) &&
+    requireNamespace("knitr") &&
+    packageVersion("knitr") >= "1.42.12")
+}
```

Comment:
This is entirely a stylistic nit, but I'd rather break the tests into individual lines and more clearly call out the return value associated with the results of each test.

```suggestion
do_use_aria <- function(knitrOptions = NULL) {
  opt <- getOption("htmlwidgets.use_aria")
  if (!is.null(opt)) return(opt)
  
  if (!requireNamespace("knitr", quietly = TRUE)) return(FALSE)
  if (packageVersion("knitr") <= "1.42.12") return(FALSE)
  
  "fig.alt" %in% names(knitrOptions)
}
```

Refactoring in this way brings up a question: does it make sense for this function to return `TRUE` if `fig.alt` isn't part of `knitrOptions`? Similarly, does it make sense to force `use_aria` even if we have an outdated version of knitr?

Reply:
The ` %||%` operator is used fairly frequently in `htmlwidgets` code, so I think it's reasonable to use it here.

`knitr` will use `fig.cap` as the alt label if `fig.alt` isn't specified but the widget asks for alt text by including the `aria-labelledby` attribute.  This shouldn't be the default, but I think users should be able to ask for it.

Similarly, an `htmlwidget` might be used in some context other than `knitr`, so I think it should be possible to force the attribute.

Reply:
Thanks for the answers; I can see how forcing `use_aria` could be useful and I ca. In addition to the global R option, we could also consult a knitr chunk option, e.g. `use_aria`, which would more easily allow users to set the option at the chunk or document level.

> The `%||%` operator is used fairly frequently in `htmlwidgets` code, so I think it's reasonable to use it here.

My comment wasn't so much about `%||%` as the entire expression. htmlwidgets is maintained by a reasonably large number of people and it's worth the tradeoff to be as readable as possible. I'm much more confident that future contributors will understand and correctly update the refactored version than if all of the tests are performed in a single expression. For example, it's much easier to correctly introduce a new check, e.g. for a `use_aria` chunk option, in the more verbose version than in the single expression version.